### PR TITLE
chore: added coverage generation stage in ci

### LINKED ISSUE #25: Adding code coverage to circle CI and modify the build badge to show the coverage
https://circleci.com/docs/2.0/code-coverage/#javascript

### PR DESCRIPTION
**Pull request by christyjacob4**
# Title
Added Coverage generation stage in CI 

## Description of what's changing
This step would be necessary to include the coverage badges in the README.md

## What else might be impacted?
None

## Which issue does this PR relate to?
Fixes #25 

## Checklist

[x] Tests are written and maintain or improve code coverage
[x] I've tested this in my application using `yarn link` (if applicable)
[x] You consent to and are confident this change can be released with no regression

### CODE REVIEW

**Comment by christyjacob4** [comment]
Screenshot 2020-10-01 at 9 35 14 AM

The artifacts are generated as part of the build process. Can I use the links of these artifacts directly in the readme for #19

**Comment by reubenae** [comment]
@christyjacob4 that looks like a good solution to me. But will defer to @skodamarthi for final approval as she is much more up to speed on this than me.

**Comment by christyjacob4** [comment]
Hi @reubenae a potential problem with this approach is that there is no way to reference the artifacts from the readme since their URLs are not fixed. I'm looking around for solutions but the suggestion is to use codecov or [Coveralls](https://dev.to/dxnchima/integrating-circleci-with-coveralls-for-your-next-node-project-hi1)

**Comment by christyjacob4** [comment]
@skodamarthi Yeap sure. if nothing works we could go with the coveralls approach

**Comment by christyjacob4** [comment]
Shall I add support for coveralls in this PR ?

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "scripts": {
     "test": "jest",
     "build": "webpack",
-    "test:coverage": "npm test -- --coverage",
+    "test:coverage": "npm test -- --coverage && cat ./coverage/lcov.info | coveralls",
     "test:badges": "npm run test:coverage  && jest-coverage-badges",
```
Comment (skodamarthi) [inline package.json:9]: @christyjacob4 If we are using coveralls for badge, I think we can get rid off jest-coverage-badges.

File: .circleci/config.yml [hunk 2]
```diff
@@ -22,6 +24,19 @@ jobs:
 
       # run tests!
       - run: npm test
+
+      # generate badges 
+      - run : 
+          name : "Generate Code Coverage and Badges"
+          command : npm run test:badges
```
Comment (skodamarthi) [inline .circleci/config.yml:31]: Can this be removed as you are setting up coveralls?

File: package.json (hunk 1 above)
Comment (christyjacob4) [reply package.json:9]: Yes let me remove "test:badges"

File: .circleci/config.yml (hunk 2 above)
Comment (christyjacob4) [reply .circleci/config.yml:31]: Yes I'll be removing it.. Doing some final tests on my fork. I'll keep you updated. I might need your help to setup coveralls for this repo. 
Also please note that coveralls provides one aggregated coverage badge and not 4 separata badges

**Comment by christyjacob4** [comment]
References : 
https://dev.to/dxnchima/integrating-circleci-with-coveralls-for-your-next-node-project-hi1
https://medium.com/@ollelauribostr/start-measuring-coverage-with-jest-travis-ci-and-coveralls-1867928aca42
https://medium.com/@okunladekayode/how-i-integrated-coveralls-with-my-nodejs-project-95e88b78ae68


I have added support for coveralls 


Few action items from the repo owners
1. Setup COVERALLS_REPO_TOKEN as an env variable in circleci
2. Log into coveralls and add coveralls to this repo. 

This is the end result.
Screenshot 2020-10-02 at 5 28 47 PM

Let me know once doe @skodamarthi

**Comment by skodamarthi** [comment]
@christyjacob4 I have done the steps. Do I also need to create coverall.yml file and add the environment var to it?

**Comment by christyjacob4** [comment]
@skodamarthi  yes you need to create a `.coveralls.yml` file locally as well `repo_token: <YOUR_REPO_TOKEN>`

Else you will be unable to push your code  because the husky `pre-push` hook runs `npm test && npm run test:coverage`

and `npm run test:coverage` requires coveralls. 

What I can do here, is to only run `npm test` in the pre-commit hook. 
Then we wouldn't need `.coveralls.yml` locally

**Comment by skodamarthi** [comment]
@christyjacob4 Sounds good! I believe this PR can be merged now without any additional changes. Right?

**Comment by christyjacob4** [comment]
Yes @skodamarthi  we're good to go 👍

**Review by skodamarthi — APPROVED** [review]

**Pull request merged**